src/main/java/com/iexec/worker/docker/DockerService.java

```diff
@@ -34,13 +34,14 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 import java.util.function.Predicate;
 
 @Slf4j
 @Service
 public class DockerService {
 
-    private final HashSet<String> runningContainersRecord;
+    private final Set<String> runningContainersRecord;
```

Comment:
Don't you think it is safer to use a thread-safe hashSet in addition to the modifications you added? Since concurrent threads will call `removeFromRunningContainersRecord()`.
```
runningContainersRecord = ConcurrentHashMap.newKeySet();
```

Reply:
I totally agree yes! It's required for both add and put :+1:

Reply:
Muchas gracias!